### PR TITLE
fix(supi): --fix-lockfile should preserve dependencies in the existing lockfile

### PR DESCRIPTION
**Pull request by wre232114**
Currently all fields in the lockfile entries will be regenerated when fixing the lockfile, including the dependencies field. This behavior  will lead to lose locked dependencies version of an existing lockfile entry, which is not work as intended.

Currently preferedVersion are used to lock the existing version and works well in most case, but in some edge cases, the lockfile structure may change due to the lose of locked dependency version info

### CODE REVIEW

File: packages/supi/src/install/index.ts [hunk 1]
```diff
@@ -728,7 +728,7 @@ const _installInContext: InstallFunction = async (projects, ctx, opts) => {
   })
   const projectsToResolve = await Promise.all(projects.map(async (project) => _toResolveImporter(project)))
 
-  // Ignore some field when fixing lockfile, so this filed can be regenereated
+  // Ignore some fields when fixing lockfile, so these fileds can be regenereated
```
Comment (zkochan) [inline packages/supi/src/install/index.ts]:
```suggestion
  // Ignore some fields when fixing lockfile, so these fields can be regenerated
```

File: .changeset/fifty-snakes-refuse.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"supi": patch
+---
+
+--fix-lockdile should preserve existing lockfile's dependencies and optionalDependencies
```
Comment (zkochan) [inline .changeset/fifty-snakes-refuse.md]:
```suggestion
`--fix-lockfile` should preserve existing lockfile's `dependencies` and `optionalDependencies`.
```

**Review by zkochan — COMMENTED** [review]
Please add a test

**Comment by wre232114** [comment]
> Please add a test

ok, I will cover it with a test case

**Review by zkochan — APPROVED** [review]

**Pull request merged**